core/src/main/java/com/avast/metrics/core/multi/GroupMonitor.java

```diff
@@ -0,0 +1,150 @@
+package com.avast.metrics.core.multi;
+
+import com.avast.metrics.TimerPairImpl;
+import com.avast.metrics.api.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Wrapper around a group of monitors.
+ */
+public class GroupMonitor implements Monitor {
+
+    private final List<Monitor> monitors;
+    private final Naming naming;
+
+    /**
+     * Factory method.
+     *
+     * @param monitors   list of monitors
+     * @return multiple  monitor containing all passed monitors
+     */
+    public static Monitor of(Monitor... monitors) {
+        return new GroupMonitor(Arrays.asList(monitors),Naming.defaultNaming());
+    }
+
+    /**
+     * Factory method.
+     *
+     * @param monitors   list of monitors
+     * @param naming     naming conventions for TimerPairs
+     * @return multiple  monitor containing all passed monitors
+     */
+    public static Monitor of(Naming naming, Monitor... monitors) {
+        return new GroupMonitor(Arrays.asList(monitors), naming);
+    }
+
+    private GroupMonitor(List<Monitor> monitors, Naming naming) {
+        if (monitors.size() == 0) {
```

Comment:
I would forbid size < 2, see e.g. `MultiCounter` that has the same condition.

Reply:
Would't be enough to show a warning instead of failing? :thinking:

Reply:
I do not see any reason why limit to >= 2 when there is no real reason for it

Reply:
@jendakol I personally prefer to fail fast on the first run during development. You can miss log message.
@hanny24 But there is **already** the limit in other classes. So either update this one or all other ones to be consistent.

```java
    public MultiCounter(List<Counter> counters) {
        if (counters.size() < 2) {
            throw new IllegalArgumentException("Multi counter from less than 2 counters makes no sense");
        }

        this.counters = counters;
    }
```

Reply:
But it makes somehow  sense in SummaryMonitor. But not really in GroupMonitor.